api/tests/unit/app_analytics/test_tasks.py

```diff
@@ -278,6 +278,74 @@ def test_populate_feature_evaluation_bucket_15m(freezer):
     assert buckets[7].total_count == 15
 
 
+@pytest.mark.freeze_time("2023-01-19T09:00:00+00:00")
+@pytest.mark.django_db(databases=["analytics"])
+def test_populate_feature_evaluation_bucket__upserts_buckets(freezer) -> None:
```

Comment:
`freezegun` upgrade is blocked by restframework/django 4 upgrade, hence no type annotation here.